Sources/Moho/ScopeDiagnostics.swift

```diff
@@ -8,26 +8,44 @@
 import Lithosphere
 
 extension Diagnostic.Message {
+  static func undeclaredIdentifier(_ n: QualifiedName) -> Diagnostic.Message {
+    return Diagnostic.Message(.error, "Use of undeclared identifier '\(n)'")
+  }
+
+  static func undeclaredIdentifier(_ n: Name) -> Diagnostic.Message {
+    return Diagnostic.Message(.error, "Use of undeclared identifier '\(n)'")
+  }
+
   static func ambiguousName(
     _ n: Name, _ others: [FullyQualifiedName]) -> Diagnostic.Message {
-    return Diagnostic.Message(.error, "\(n) could be any of \(others)")
+    return Diagnostic.Message(.error, "'\(n)' could be any of \(others)")
   }
 
   static func nameReserved(_ n: Name) -> Diagnostic.Message {
-    return Diagnostic.Message(.error, "cannot use reserved name \(n)")
+    return Diagnostic.Message(.error, "cannot use reserved name '\(n)'")
   }
 
   static func nameShadows(_ n: Name) -> Diagnostic.Message {
-    return Diagnostic.Message(.error, "cannot shadow name \(n)")
+    return Diagnostic.Message(.error, "cannot shadow name '\(n)'")
   }
 
   static func nameShadows(
     _ n: Name, local: FullyQualifiedName) -> Diagnostic.Message {
     return Diagnostic.Message(.error,
-                              "cannot shadow qualified name \(local) with \(n)")
+      "cannot shadow qualified name '\(local)' with '\(n)'")
```

Comment:
Rather than "cannot" for all of these diagnostics, we should just say "name shadows qualified name '\(local)' with '\(n)', and let the fact that it's an error communicate that they can't do that.